arrows/ffmpeg/ffmpeg_video_input.h

```diff
@@ -27,6 +27,12 @@ class KWIVER_ALGO_FFMPEG_EXPORT ffmpeg_video_input
   : public vital::algo::video_input
 {
 public:
+  enum seek_mode {
+    SEEK_MODE_EXACT,
+    SEEK_MODE_KEYFRAME_BEFORE,
+    SEEK_MODE_KEYFRAME_AFTER,
```

Comment:
Correct me if I am wrong, but currently these options are hidden from the user. If they are ever exposed via the configuration we should insure they are all tested. I assume the non-default option of SEEK_MODE_KEYFRAME_BEFORE is effectively tested in `test_ffmpeg_video_input_clip`.

Reply:
You bring up a good point. These are in the implementation-specific header, so they are hidden from the user in the sense of not being accessible from a generic `video_input`. However, they are publicly accessible from an `ffmpeg_video_input` (and they need to be, for `ffmpeg_video_input_clip` to use them). `SEEK_MODE_EXACT` is the default behavior, and passes existing tests. `SEEK_MODE_KEYFRAME_BEFORE` is used in, and indirectly tested via, `ffmpeg_video_input_clip`. But `SEEK_MODE_KEYFRAME_AFTER` is not tested anywhere, nor is it used (currently) in `ffmpeg_video_input_clip`. I just included it for completeness. So I guess there are a few options here:

1. Leave as-is because none of this is accessible from `video_input`.
2. Add `ffmpeg_video_input` tests for `KEYFRAME_AFTER`.
3. Implement `KEYFRAME_AFTER` behavior in `ffmpeg_video_input_clip`, and add a test there.
4. Remove `KEYFRAME_AFTER` for now because there's no pressing reason to have it and it complicates the code.

Any thoughts on which is best?

Reply:
@chetnieter (bump)

Reply:
Let's go with 4. Correct if I am wrong, but it that option becomes necessary in the future, it is relatively trivial to add it back.

Reply:
Sounds good, will do. I've retained the current version in a local branch in case it comes up again in the future.